runtime/gc_structs/ClassArrayClassSlotIterator.cpp

```diff
@@ -36,32 +36,36 @@
  * @return the next slot containing an object reference
  * @return NULL if there are no more such slots
  */
-J9Class **
+J9Class *
 GC_ClassArrayClassSlotIterator::nextSlot() 
 {
-	J9Class **slotPtr;
-
-	switch(_state) {
-	case classArrayClassSlotIterator_state_arrayClass:
-		slotPtr = (J9Class **)&_iterateClazz->arrayClass;
-		if(!_isArrayClass) {
-			_state = classArrayClassSlotIterator_state_done;
-		} else {
+	while (_state != classArrayClassSlotIterator_state_done) {
+		J9Class *classPtr = NULL;
+		switch(_state) {
```

Comment:
since you fixed it at other spots, put extra space before oval bracket. here, and down below for 'if'